OrbitClientServices/ProcessManager.cpp

```diff
@@ -124,11 +126,16 @@ void ProcessManagerImpl::WorkerFunction() {
       continue;
     }
 
-    absl::MutexLock callback_lock(&mutex_);
-    const auto& processes = result.value();
-    process_list_.assign(processes.begin(), processes.end());
-    if (process_list_update_listener_) {
-      process_list_update_listener_(this);
+    {
+      absl::MutexLock process_lock(&process_list_mutex_);
+      const auto& processes = result.value();
+      process_list_.assign(processes.begin(), processes.end());
+    }
+    {
+      absl::MutexLock callback_lock(&process_list_update_listener_mutex_);
+      if (process_list_update_listener_) {
+        process_list_update_listener_(this);
+      }
```

Comment:
I am guessing the problem here is that listener may call GetProcessList which will lead to recursive lock. Have you considered passing reference to process_list_ here maybe?

Reply:
I like this Idea as well. Then I would get rid of the function GetProcessList() altogether, which requires a small change in App.cpp where GetProcessList() is used. 
I also dont mind how the change is right now.

Reply:
Good point. I changed that, PTAL. Nevertheless, I think it's a good idea to keep this as two mutexes, since they guard different objects, and it seems excessive to block the whole process list update during the callback when the callback operates on a copy anyways.